flow/Platform.cpp

```diff
@@ -1666,6 +1666,27 @@ void atomicReplace( std::string const& path, std::string const& content, bool te
 		f = textmode ? fopen( tempfilename.c_str(), "wt" ) : fopen(tempfilename.c_str(), "wb");
 		if(!f)
 			throw io_error();
+	#ifdef _WIN32
+	#elif defined(__unixish__)
+		// get the uid/gid/mode bits of old file and set it on new file, else fail
+		struct stat info;
+		if (stat(path.c_str(), &info) < 0) {
+			TraceEvent("StatFailed").detail("path", path);
+			throw io_error();
+		}
+		if (chown(tempfilename.c_str(), info.st_uid, info.st_gid) < 0) {
+			deleteFile(tempfilename);
+			TraceEvent("ChownFailed").detail("tempfilename", tempfilename).detail("uid", info.st_uid).detail("gid", info.st_gid);
```

Comment:
Since delete can throw errors, I think we should move the trace event above `deleteFile` so as not to lose it. I'd probably throw `path` in there too just so we know what file we failed to replace. The same applies for the block below as well.

Also, the format that we use for trace event details is to capitalize the first letter and use camel case. For this trace event, that would then be `TempFilename`, `Path`, `Uid`, and `Gid`.

Reply:
Sounds good, will address these comments.